### PR TITLE
add comments on grid components

### PR DESCRIPTION
**Pull request by simonbyrne**
### Description

Since these aren't documented anywhere, it is useful to at least have comments

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Numerics/Mesh/Grids.jl [hunk 1]
```diff
@@ -83,19 +83,22 @@ const vgeoid = (
...
-# The MH terms are for integrating over a plane.
 const _nsgeo = 5
 const _n1, _n2, _n3, _sM, _vMI = 1:_nsgeo
-const sgeoid = (n1id = _n1, n2id = _n2, n3id = _n3, sMid = _sM, vMIid = _vMI)
+const sgeoid = (
+    n1id = _n1,  # outward normal = (n1,n2,n3)
+    n2id = _n2,
+    n3id = _n3,
+    sMid = _sM,
+    vMIid = _vMI)
```
Comment (simonbyrne) [inline src/Numerics/Mesh/Grids.jl:101]: Would like to expand on these as well.

File: src/Numerics/Mesh/Grids.jl [hunk 2]
```diff
@@ -83,19 +83,23 @@ const vgeoid = (
     ξ1x3id = _ξ1x3,
     ξ2x3id = _ξ2x3,
     ξ3x3id = _ξ3x3,
-    Mid = _M,
-    MIid = _MI,
-    MHid = _MH,
+    Mid = _M,    # J * ωᵢ * ωⱼ * ωₖ, where ωᵢ are quadrature weights, J is det(∂x/∂ξ)
```
Comment (skandalaCLIMA) [inline src/Numerics/Mesh/Grids.jl:86]: # J .* ( ωₖ (kron) ωⱼ (kron) ωᵢ) ,

**Comment by sandreza** [comment]
I'm not sure this is helpful, but it seems like the right place to post this.
These are some of the notes I wrote for myself a while back (I'm not sure if that has changed since the FV update)

# the grid data structure contains 37 members
# (6) Surface structures
# sgeo, and the 5 ids
# (17) Volume structures
# vgeo, and the 16 ids
# (2) Volume to Surface index
# vmap⁻, vmap⁺
# (1) Topology
# topology
# (3) Operators
# grid.D (differentiation matrix), grid.Imat (integration matrix?), grid.ω (quadrature weights)
# information (1)
# activedofs (active degrees of freedom)
# Domain Boundary Information (1)
# elemtobndy (element to boundary)
# MPI stuff (6)
# interiorelems, exteriorelems, nabrtovmaprecv, nabrtovmapsend, vmaprecv, vmapsendv 

# sgeo, the first index is for the ids
# the second index are the gauss-lobatto entries
# the third index is the face
# the last index is the element
# the ids are: (1, n1id), (2 n2id), (3 n3id) (4, sMid), (5, vMIid)
# grid.sgeo[1,:,:,:] are normals in the n1 direction
# norm(grid.vgeo[:, grid.MIid, :][grid.vmap⁻] - grid.sgeo[5,:,:,:]) is zero
# x[grid.vmap⁻[10]] is connected to x[grid.vmap⁺[10]]

# vgeo, first index is guass-lobatto points
# second index is ids
# third index is elements
# the are are 16 ids
# 1-9 are metric terms
# 10 is the mass matrix
# 11 is the inverse mass matrix
# 12 is MHid (horizontal mass matrix?)
# 13-15 are x1 x2 x3
# 16 is the vertical volume jacobian
# grid.vgeo[:, grid.ξ1x1id, :]

**Comment by sandreza** [comment]
I'm not quite sure if this is the right place to post this, but I wrote these comments for myself a while back

```julia
# the grid data structure contains 37 members
# (6) Surface structures
# sgeo, and the 5 ids
# (17) Volume structures
# vgeo, and the 16 ids
# (2) Volume to Surface index
# vmap⁻, vmap⁺
# (1) Topology
# topology
# (3) Operators
# grid.D (differentiation matrix), grid.Imat (integration matrix?), grid.ω (quadrature weights)
# information (1)
# activedofs (active degrees of freedom)
# Domain Boundary Information (1)
# elemtobndy (element to boundary)
# MPI stuff (6)
# interiorelems, exteriorelems, nabrtovmaprecv, nabrtovmapsend, vmaprecv, vmapsendv 

# sgeo, the first index is for the ids
# the second index are the gauss-lobatto entries
# the third index is the face
# the last index is the element
# the ids are: (1, n1id), (2 n2id), (3 n3id) (4, sMid), (5, vMIid)
# grid.sgeo[1,:,:,:] are normals in the n1 direction
# norm(grid.vgeo[:, grid.MIid, :][grid.vmap⁻] - grid.sgeo[5,:,:,:]) is zero
# x[grid.vmap⁻[10]] is connected to x[grid.vmap⁺[10]]

# vgeo, first index is guass-lobatto points
# second index is ids
# third index is elements
# the are are 16 ids
# 1-9 are metric terms
# 10 is the mass matrix
# 11 is the inverse mass matrix
# 12 is MHid (horizontal mass matrix?)
# 13-15 are x1 x2 x3
# 16 is the vertical volume jacobian
# grid.vgeo[:, grid.ξ1x1id, :]
```

**Comment by jkozdon** [comment]
see https://github.com/CliMA/ClimateMachine.jl/pull/1950

**Comment by jkozdon** [comment]
> see #1950

Which is a PR into this branch

File: src/Numerics/Mesh/Grids.jl (hunk 1 above)
Comment (valeriabarra) [reply src/Numerics/Mesh/Grids.jl]: Jeremy too care of it in https://github.com/CliMA/ClimateMachine.jl/pull/1950/

**Review by charleskawczynski — APPROVED** [review]

**Review by jkozdon — APPROVED** [review]

**Review by mwarusz — APPROVED** [review]
LGTM

**Review by valeriabarra — APPROVED** [review]

**Comment by skandalaCLIMA** [comment]
src/Numerics/Mesh/Grids.jl line:92
I believe this should be J .* ( ω₃ ⊗ ω₂ ⊗ ω₁) instead of #    J .* (ωᵢ ⊗ ωⱼ ⊗ ωₖ)? Please advise.
Since ω₁ is the “fastest” direction right? (Here, 1, 2, 3 refer to \xi_1, \xi_2, \xi_3 reference directions)
(ωᵢ ⊗ ωⱼ ⊗ ωₖ) will flip the order in the kronecker product.

**Pull request merged**